ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+@dataclass
+class TVSwinTransformerEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["swin_transformer_torch"],
+        default="swin_transformer_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    layer_norm_eps: float = schema_utils.NonNegativeFloat(
-        default=1e-12,
-        description="The epsilon used by the layer normalization layers.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["layer_norm_eps"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "t",
+            "s",
+            "b",
+        ],
+        default="t",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVSwinTransformerEncoder"]["model_variant"],
     )
 
-    gradient_checkpointing: bool = schema_utils.Boolean(
-        default=False,
-        description="",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["gradient_checkpointing"],
+
+@register_encoder_config("vit_torch", IMAGE)
+@dataclass
+class TVViTEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["vit_torch"],
+        default="vit_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    patch_size: int = schema_utils.PositiveInteger(
-        default=16,
-        description="The image patch size. Each patch is patch_size² pixels. Must evenly divide the image width and "
-        "height.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["patch_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "b_16",
+            "b_32",
+            "l_16",
+            "l_32",
+            "h_14",
+        ],
+        default="b_16",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVViTEncoder"]["model_variant"],
     )
 
-    saved_weights_in_checkpoint: bool = schema_utils.Boolean(
-        default=False,
-        description="Are the pretrained encoder weights saved in this model's checkpoint? Automatically set to"
-        "True for trained models to prevent loading pretrained encoder weights from model hub.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["saved_weights_in_checkpoint"],
+
+@register_encoder_config("vgg_torch", IMAGE)
+@dataclass
+class TVVGGEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["vgg_torch"],
+        default="vgg_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    trainable: bool = schema_utils.Boolean(
-        default=True,
-        description="Is the encoder trainable.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["trainable"],
+    model_variant: Union[int, str] = schema_utils.OneOfOptionsField(
```

Comment:
By default the `OneOfOptionsField` allows `None` as a valid value. If this is intended, please add `None` to the `Union` annotation.

Reply:
`None` is not allowed, so added the parameter `allow_none=False`.